### DISCUSSION TITLE
How do I override the name of the random device block on `unlock`?

### DISCUSSION THREAD

**Discussion by andradei** [storaged-project/udisks#1303] [Q&A] [answered]
I want to `udisksctl unlock --block-device /dev/sda1 my_custom_drive` so it is mapped to `my_custom_drive` instead or the random `luks-12341234-21341243-afafaf-123414`.

How can I achieve that?

**Comment by vojtechtrefny** [comment, marked as answer]
Currently the only way is to do this is to have the device referenced in `/etc/crypttab`, we will use the name specified there.

We could do something similar we do for mounting where we use filesystem label for mountpoint and use LUKS label here, that would be a relatively simple change.

**Reply by andradei** [reply]
That's an interesting solution.
I'm learning about mounting, partitions, filesystems, LUKS, etc. and didn't know bout `/etc/crypttab` so thank you for the answer, that worked.
